### ISSUE TITLE
No space left on device while building SD card image

### ISSUE THREAD

**Issue by ghostbuster91** [nakato/nixos-bpir3-example#3]
Hi, I flashed your image on the SD card and booted BPI from it. 

Then I wanted to build my own image using your flake and recent changes. I cloned the repository and run the command from the readme. 

Unfortunately it failed with a following error:
```
linux> fixdep: not all data was written to the output
linux> make[2]: *** [/nix/store/nmkqiyicgv8636g6fkv1gpw3kzsxq8nm-linux-6.4.0-rc5-dev/lib/modules/6.4.0-rc5/source/scripts/Makefile.host:127: scripts/kconfig/util.o] Error 1
linux> make[2]: *** Deleting file 'scripts/kconfig/util.o'
linux> make[1]: *** [/nix/store/nmkqiyicgv8636g6fkv1gpw3kzsxq8nm-linux-6.4.0-rc5-dev/lib/modules/6.4.0-rc5/source/Makefile:692: oldconfig] Error 2
linux> make: *** [Makefile:226: __sub-make] Error 2
uboot-mt7986a_bpir3_sd_defconfig> FATAL ERROR: Error writing device tree blob: No space left on device
uboot-mt7986a_bpir3_sd_defconfig> Check /build/u-boot-2023.07-rc3/arch/arm/dts/.mt7623a-unielec-u7623-02-emmc.dtb.pre.tmp for errors
uboot-mt7986a_bpir3_sd_defconfig> make[2]: *** [scripts/Makefile.lib:335: arch/arm/dts/mt7623a-unielec-u7623-02-emmc.dtb] Error 1
uboot-mt7986a_bpir3_sd_defconfig> make[2]: *** Deleting file 'arch/arm/dts/mt7623a-unielec-u7623-02-emmc.dtb'
uboot-mt7986a_bpir3_sd_defconfig> make[2]: *** Waiting for unfinished jobs....
error: builder for '/nix/store/8wfzrm14m6iphdc9mm27mmjban89v9rf-linux-6.4.0-rc5.drv' failed with exit code 2
       note: build failure may have been caused by lack of free disk space
error: 1 dependencies of derivation '/nix/store/cl7p91bhs4z4p5lkmxn5rh58y0dbqand-nixos-system-bpir3-23.05.20230506.897876e.drv' failed to build
error: 1 dependencies of derivation '/nix/store/y3m2hs8iybr1kj7dplafx05h5vksp432-nixos-bananapir3-sd.drv' failed to build
```

Any idea why that happened? My SD card is 120GB and I don't have nvme disk. 
Here is an output from `df`: 
```
 df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs        100M     0  100M   0% /dev
tmpfs           995M     0  995M   0% /dev/shm
tmpfs           498M  3.0M  495M   1% /run
tmpfs           995M  384K  995M   1% /run/wrappers
/dev/mmcblk0p6  117G  5.3G  107G   5% /
/dev/mmcblk0p5  511M   31M  481M   7% /boot
tmpfs           199M   12K  199M   1% /run/user/0
```

**Comment by nakato** [comment]
That's odd, that should be substantially more space than required to build an updated SD image.

I've downloaded my image back off github, flashed it to a 128GB SD card, and booted it on my second unit, running the following.

```
[root@bpir3:~]# nix --extra-experimental-features nix-command --extra-experimental-features flakes shell nixpkgs#git

[root@bpir3:~]# git clone https://github.com/nakato/nixos-bpir3-example
Cloning into 'nixos-bpir3-example'...
remote: Enumerating objects: 77, done.
remote: Counting objects: 100% (77/77), done.
remote: Compressing objects: 100% (51/51), done.
remote: Total 77 (delta 34), reused 64 (delta 21), pack-reused 0
Receiving objects: 100% (77/77), 53.55 KiB | 4.87 MiB/s, done.
Resolving deltas: 100% (34/34), done.

[root@bpir3:~]# cd nixos-bpir3-example/

[root@bpir3:~/nixos-bpir3-example]# git rev-parse HEAD
a012b6e58353412b7af4d351c2da6e32c06075cc

[root@bpir3:~/nixos-bpir3-example]# nix --extra-experimental-features nix-command --extra-experimental-features flakes build  -L '.#nixosConfigurations.bpir3.config.system.build.sdImage'
```

That'll take a few hours, but I'll just let it run and see if it reproduces.


What does the output of `df -i` show?  If it's not diskspace, it's possible that the filesystem ran out of inodes, but that shouldn't be the case after the resize.  Just to confirm, nothing's been mounted to `/tmp` before attempting to build?

**Comment by ghostbuster91** [comment]
I ran your commands and it seems fine now. In my case I couldn't figure out how to add git to clone the repository (`nix-shell -p git` didn't work as there was no `NIX_PATH` set). I ended up adding a channel which solved problem with getting git. But it seems that there is a difference and using channel doesn't work. 


Thanks for the quick reply and putting that flake together, I have almost given up on running nixos on bpir3. If there is any way I could buy you a digital coffee I would be happy to do so :)

After building the new image I did `nixos-rebuild switch --flake .` and it worked like a charm!! :tada:
